File: Projeto_5/src/SensorDiagnostico.cpp
#include "SensorDiagnostico.h"
#include <iostream>

SensorDiagnostico::SensorDiagnostico(std::string id, int ts, int rpm, double temp) 
    : Dispositivo(id, ts), rpmMotor(rpm), temperaturaFluido(temp) {}

void SensorDiagnostico::processarDados() {
    std::cout << "[DIAGNÓSTICO] ID: " << idDispositivo << " | Tempo: " << timestamp 
              << " | Motor: " << rpmMotor << " RPM | Temp: " << temperaturaFluido << "°C" << std::endl;
}

File: Projeto_5/models/SensorDiagnostico.py
from models.Dispositivo import Dispositivo

class SensorDiagnostico(Dispositivo):
    def __init__(self, id_dispositivo: str, timestamp: int, rpm_motor: int, temperatura_fluido: float):
        Dispositivo.__init__(self, id_dispositivo, timestamp)
        self.__rpmMotor = rpm_motor                  
        self.__temperaturaFluido = temperatura_fluido  

    def processarDados(self) -> None:
        print(f"[ RODANDO... ] Processando {self._idDispositivo} | RPM: {self.__rpmMotor} | Temp: {self.__temperaturaFluido}C")